### DISCUSSION TITLE
Question on backend structure for booking app

### DISCUSSION THREAD

**Discussion by maurijr8** [supabase/supabase#20039] [Questions]
hi everyone! My apologies if this is not the right forum. I just started learning a little bit more about Supabase (and backend in general) and would love to hear from more experienced users.

I have a quick question on a booking platform that I am building and would love to see if I can get some help/ideas here:

**Context:** I am using Supabase and a no-code UI builder to work on a barber's booking app. I have a table for bookings, barbers, services, and hours of operation.

**Question:** What would be the best way to fetch the available booking slots based on different things:
- Date picked in the UI
- Available slots based on the hours of operation table
- Interval of each slot (e.g., 60min, 30min, etc.)
- Taken slots from the bookings table
- Duration of the service

I am pretty sure I will need some custom functions here, but not sure if I should do this in Supabase or the UI builder. I believe I should do it in Supabase, but just not even sure how to get it started. 😅

Thanks so much in advance!

**Comment by GaryAustin1** [comment]
This is a pretty broad question.  I would research `Postgres example booking sites` in Google or even try an AI with some questions qualifying with Postgres.  I don't think this is anything Supabase specific but just a matter of deciding on tables and queries to do what you want.  These are mainly Postgres issues.
